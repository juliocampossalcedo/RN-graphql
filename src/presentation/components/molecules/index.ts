export { default as MDetail } from './mDetail/detail';