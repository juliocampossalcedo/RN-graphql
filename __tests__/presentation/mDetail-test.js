import { MDetail } from '../../src/presentation/components/molecules/index';
import renderer from 'react-test-renderer';

describe('Test render of image and texts', () => {
    it('it should renders correctly', () => {
        //this way of testing consists in create an Snapshot of the rendered component 
        //so if it change in the future this test will fail because the rendered component
        // should be different of its previous version
        const mDetailSnap = renderer.create(
            <MDetail />
        ).toJSON();  
        expect(mDetailSnap).toMatchSnapshot();
    });
});
