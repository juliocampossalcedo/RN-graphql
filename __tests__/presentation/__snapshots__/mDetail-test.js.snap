// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render of image and texts it should renders correctly 1`] = `
<View>
  <Image
    source={
      {
        "uri": "",
      }
    }
    style={
      [
        {
          "height": 750,
          "marginBottom": 50,
          "width": 750,
        },
      ]
    }
  />
  <Text
    style={
      [
        {
          "fontSize": 20,
          "fontWeight": "500",
          "paddingBottom": 20,
          "textAlign": "center",
        },
      ]
    }
  >
    Name: 
  </Text>
  <Text
    style={
      [
        {
          "fontSize": 20,
          "fontWeight": "500",
          "paddingBottom": 20,
          "textAlign": "center",
        },
      ]
    }
  >
    Status: 
  </Text>
  <Text
    style={
      [
        {
          "fontSize": 20,
          "fontWeight": "500",
          "paddingBottom": 20,
          "textAlign": "center",
        },
      ]
    }
  >
    Species: 
  </Text>
  <Text
    style={
      [
        {
          "fontSize": 20,
          "fontWeight": "500",
          "paddingBottom": 20,
          "textAlign": "center",
        },
      ]
    }
  >
    Gender: 
  </Text>
  <Text
    style={
      [
        {
          "fontSize": 20,
          "fontWeight": "500",
          "paddingBottom": 20,
          "textAlign": "center",
        },
      ]
    }
  >
    Origin: 
  </Text>
</View>
`;
